x-pack/platform/plugins/shared/fleet/public/applications/integrations/sections/epm/screens/installed_integrations/components/installed_integrations_search_bar.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFilterButton,
+  EuiFilterGroup,
+  useEuiTheme,
+  EuiFieldSearch,
+} from '@elastic/eui';
+import React, { useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { css } from '@emotion/react';
+import { i18n } from '@kbn/i18n';
+import useDebounce from 'react-use/lib/useDebounce';
+
+import type { InstalledIntegrationsFilter, PackageInstallationStatus } from '../types';
+import { useAddUrlFilters } from '../hooks/use_url_filters';
+
+import { InstalledIntegrationsActionMenu } from './installed_integration_action_menu';
+
+const SEARCH_DEBOUNCE_MS = 250;
+
+export const InstalledIntegrationsSearchBar: React.FunctionComponent<{
+  filters: InstalledIntegrationsFilter;
+  countPerStatus: { [k: string]: number | undefined };
+  customIntegrationsCount: number;
+}> = ({ filters, countPerStatus, customIntegrationsCount }) => {
+  const addUrlFilter = useAddUrlFilters();
+  const theme = useEuiTheme();
+  const [searchTerms, setSearchTerms] = useState(filters.q);
+
+  useDebounce(
+    () => {
+      addUrlFilter({
+        q: searchTerms,
+      });
+    },
+    SEARCH_DEBOUNCE_MS,
+    [searchTerms]
+  );
+  const statuses: Array<{
+    iconType: string;
+    iconColor: string;
+    status: PackageInstallationStatus;
+    label: React.ReactElement;
+  }> = useMemo(
+    () => [
+      {
+        iconType: 'warning',
+        iconColor: theme.euiTheme.colors.textWarning,
+        status: 'upgrade_available',
+
+        label: (
+          <FormattedMessage
+            id="xpack.fleet.epmInstalledIntegrations.upgradeAvailableFilterLabel"
+            defaultMessage="Upgrade"
```

Comment:
I think `Upgrade available` would be more expressive, though it might take up too much space.

Reply:
That a good point noting this to do a design review with christian when this is technically done.